pkg/cloud/aws/services/userdata/node.go

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2018 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userdata
+
+const (
+	nodeBashScript = `{{.Header}}
+
+HOSTNAME="$(hostname -f 2>/dev/null || curl http://169.254.169.254/latest/meta-data/local-hostname)"
+
+cat >/tmp/kubeadm-node.yaml <<EOF
+---
+apiVersion: kubeadm.k8s.io/v1alpha2
+kind: NodeConfiguration
+token: {{.BootstrapToken}}
```

Comment:
Will need some additional config for either the CA Cert Hashes or to ignore the CA cert here, should also probably move the endpoint configuration in here as well.

https://github.com/kubernetes/kubernetes/blob/release-1.12/cmd/kubeadm/app/apis/kubeadm/v1alpha3/types.go#L295

Reply:
I opted out for the discoveryfile, by passing in the CA certificate, what do you think?

Reply:
Did that change get pushed up yet? I'm still seeing the discoveryFile set